src/vm/database/clarity_db.rs

```diff
@@ -653,4 +782,43 @@ impl<'a> ClarityDatabase<'a> {
         let key = ClarityDatabase::make_key_for_account_nonce(principal);
         self.put(&key, &nonce);
     }
+
+    pub fn get_account_stx_locked(&mut self, principal: &PrincipalData) -> u128 {
+        let key = ClarityDatabase::make_key_for_account_stx_locked(principal);
+        let result = self.get(&key);
+        match result {
+            None => 0,
+            Some(locked) => locked
+        }
+    }
+
+    pub fn set_account_stx_locked(&mut self, principal: &PrincipalData, amount: u128) {
+        let key = ClarityDatabase::make_key_for_account_stx_locked(principal);
+        self.put(&key, &amount);
+    }
+
+    pub fn get_account_unlock_height(&mut self, principal: &PrincipalData) -> u64 {
+        let key = ClarityDatabase::make_key_for_account_unlock_height(principal);
+        let result = self.get(&key);
+        match result {
+            None => 0,
+            Some(unlock_height) => unlock_height
+        }
+    }
+
+    pub fn set_account_unlock_height(&mut self, principal: &PrincipalData, unlock_height: u64) {
+        let key = ClarityDatabase::make_key_for_account_unlock_height(principal);
+        self.put(&key, &(unlock_height as u128));
+    }
+}
+
+// access burnchain state
+impl <'a> ClarityDatabase<'a> {
+    pub fn get_burn_block_height(&self, bhh: &BurnchainHeaderHash) -> Option<u32> {
```

Comment:
Where are these functions used?

Reply:
Nowhere yet.  This aspect of the PR -- the creation of `BurnStateDB` and access to the burnchain block height and hashes -- is residual from when an earlier version of the PoX contract that needed access to the SortitionDB in order to tell what reward cycle it was currently in.  While this is no longer a requirement for this PR, I left in the `BurnStateDB` definition because we will need it imminently anyway, in order to e.g. validate Bitcoin transactions.

I can remove all instances of `BurnStateDB` if you want, and add them all back later, but this will take time since it touches  a lot of places (simply because it has to be used wherever you use `&dyn HeadersDB`).  Please advise.

Reply:
Yep -- that's what I figured. I think leaving them in is a good idea, since the BurnStateDB is going to be needed for exposing burnchain state anyways.